src/index.html

```diff
@@ -7,6 +7,43 @@
     <link rel="stylesheet" href="./styles/main.scss">
   </head>
   <body>
-    <h1>Calendar</h1>
+    <section class="calendar calendar--29">
+      <div
+        class="calendar__item
+        calendar__item--1
+        calendar__item--sun"
```

Comment:
modifier for first month day should be for the calendar block.
name modifiers as in task description.
calendar__item--1, calendar__item--2...  and for other days are redundant.

```suggestion
    <section class="calendar calendar__start-day-sun calendar__month-length--29">
      <div
        class="calendar__item"
```